### PR TITLE
Update to latest netstack to get proper resource cleanup

### PR DESCRIPTION
**Pull request by oxtoacart**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -8,14 +8,15 @@ require (
...
+	github.com/google/go-cmp v0.3.1 // indirect
+	github.com/google/netstack v0.0.0-20191116005144-95bf25ab4723
+	github.com/songgao/water v0.0.0-20190725173103-fd331bda3f4b
 	github.com/stretchr/testify v1.3.0
+	golang.org/x/sys v0.0.0-20191120155948-bd437916bb0e // indirect
+	golang.org/x/time v0.0.0-20191024005414-555d28b269f0 // indirect
 )
 
-replace github.com/google/netstack => github.com/getlantern/netstack v0.0.0-20190313202628-8999826b821d
+replace github.com/google/netstack => github.com/getlantern/netstack v0.0.0-20191210203454-f54213f594e9
```
Comment (oxtoacart) [inline go.mod]: The only reason we still have a fork is that google's netstack doesn't support 32 bit platforms. See https://github.com/getlantern/netstack/commit/f54213f594e96f64375765f7dfe5557f77735f22

File: tun.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package ipproxy
+
+import (
+	"io"
+	"os/exec"
+	"strconv"
+
+	"github.com/getlantern/errors"
+	"github.com/songgao/water"
```
Comment (oxtoacart) [inline tun.go]: gotun was kind of flaky on Linux so I switched to water which seems to be working well.

File: demo/demo.go [hunk 3]
```diff
@@ -1,3 +1,31 @@
...
+// for proxy traffic via the default gateway, like so:
+//
+// sudo route add 67.205.172.79 192.168.1.1
+//
+// Now your network traffic will route through here to your proxy.
+//
+// When you're finished, you can fix your routing table with:
+//
+// sudo route delete default
+// sudo route add default 102.168.1.1
```
Comment (hwh33) [inline demo/demo.go]: Should this be 192.168.1.1?

**Comment by hwh33** [comment]
I'm getting failing tests:

```
➜  ipproxy git:(ox/update-netstack) go test -c
➜  ipproxy git:(ox/update-netstack) sudo ./ipproxy.test
DEBUG ipproxy: if.go:79 Created TUN device named utun2
--- FAIL: TestTCPAndUDP (0.00s)
    ipproxy_test.go:109: failed to configure tun device address: ifconfig: ioctl (SIOCAIFADDR): Destination address required

DEBUG ipproxy: if.go:79 Created TUN device named utun2
--- FAIL: TestCloseCleanup (0.00s)
    ipproxy_test.go:109: failed to configure tun device address: ifconfig: ioctl (SIOCAIFADDR): Destination address required

FAIL
```

I tried running the tests on master to see if this was a regression, but I actually can't get the tests to build on master.

**Comment by hwh33** [comment]
Do you have the same issues @oxtoacart?

File: README.md [hunk 4]
```diff
@@ -10,4 +10,6 @@ This library uses Go modules. When running commands like `go test` in this repos
 
 Tests in this package require root access. The easiest way to test is to compile the tests with `go test -c` and run the output binary using the `sudo` command.
 
-Be careful if you choose to run the Go tool with the sudo command (e.g. `sudo go test`). This can cause issues if the tool attempts to download missing dependencies. Namely, the Go tool may not be able to download anything as Git will likely be using a different SSH keypair (or no keypair at all). Worse, the Go tool may create folders in $GOPATH/pkg/mod/cache owned by the root user. This can disrupt future use of the Go tool, even outside of this repository.
\ No newline at end of file
+Tests currently only run on Linux.
```
Comment (hwh33) [inline README.md]: Aha.  Sorry!

File: demo/demo.go (hunk 3 above)
Comment (oxtoacart) [reply demo/demo.go]: Yes, will fix.

File: README.md (hunk 4 above)
Comment (oxtoacart) [reply README.md]: Yeah, sorry about that, it wasn't clear. Tests should be passing now if you try on Linux.

**Comment by oxtoacart** [comment]
Thanks @hwh33 !

**Pull request merged**